src/openforms/submissions/tests/test_submission_report.py

```diff
@@ -98,6 +100,167 @@ def test_report_generation(self, mock_request):
         self.assertTrue(report.content.name.endswith("Test_Form.pdf"))
 
     def test_report_is_generated_in_same_language_as_submission(self):
+        # fixture_data
+        fields = [
+            # Component.type, user_input
+            ("bsn", "111222333"),
+            ("checkbox", "yes"),
+            ("currency", "1100000"),
+            ("date", "1911-06-29"),
+            ("email", "hostmaster@example.org"),
+            ("file", "download(2).pdf"),
+            ("iban", "NL56 INGB 0705 0051 00"),
+            ("licenseplate", "AA-00-13"),
+            ("map", "52.193459, 5.279538"),
+            ("number", "1"),
+            ("password", "Panda1911!"),  # XXX Why is this widget even an option?
+            ("phonenumber", "+49 1234 567 890"),
+            ("postcode", "3744 AA"),
+            ("radio", "Radio number one"),
+            ("select", "A fine selection"),
+            ("selectboxes", "This; That; The Other"),
+            ("signature", SIGNATURE),
+            ("textarea", "Largish predetermined ASCII"),
+            ("textfield", "Short predetermined ASCII"),
+            ("time", "23:50"),
+            ("updatenote", "C#"),
+        ]
+        submission_data = {}
+        translations = {}
+        components = []
+
+        # carthesian products are big, let's loop to fill these
+        for component_type, user_input in fields:
+            # common base
+            component = {
+                "type": component_type,
+                "key": f"{component_type}-key",
+                "label": f"Untranslated {component_type.title()} label",
+                "showInPDF": True,
+                "hidden": False,
+            }
+            submission_data[f"{component_type}-key"] = user_input
+            translations[
+                f"Untranslated {component_type.title()} label"
+            ] = f"Translated {component_type.title()} label"
+
+            # add the component
+            components.append(component)
+
+            # but massage the weirder components into shape
+            match component:
+                case {"type": "file", "key": key}:
+                    submission_data[key] = [{"originalName": "download(2).pdf"}]
+                case {"type": "radio", "key": key}:
+                    component["values"] = [
+                        {"label": "Untranslated Radio option one", "value": "radioOne"},
+                        {"label": "Untranslated Radio option two", "value": "radioTwo"},
+                    ]
+                    submission_data[key] = "radioOne"  # value of the selected label
+                    translations["Untranslated Radio option one"] = "Radio number one"
```

Comment:
I think I'd rather see a single literal dict defined outside of the loop that accounts for all the translatable strings - it is closer to Formio's model that you can only have a single key-value map for the translations and that it's _not_ at the component level.

And following my own reasoning, it'd probably make more sense to set up `submission_data` by hand too instead of doing that programmatically.